pkg/repository/tenant.go

```diff
@@ -71,27 +112,188 @@ func (t *tenantRepository) get(ctx context.Context, id string) (*mdcv1.Tenant, e
+	}
+
+	var labels []string
+	if tenant.Meta != nil && tenant.Meta.Labels != nil && len(tenant.Meta.Labels.Labels) > 0 {
+		labels = tag.TagMap(tenant.Meta.Labels.Labels).Slice()
+	}
+
+	return &mdcv1.Tenant{
+		Meta: &mdcv1.Meta{
+			Id:          tenant.Login,
+			Kind:        "Tenant",
+			Annotations: ann,
+			Labels:      labels,
+		},
+		Name:        tenant.Name,
+		Description: tenant.Description,
+	}, nil
+}
+
+func (te *tenantRepository) convertToProto(t *mdcv1.Tenant) (*apiv2.Tenant, error) {
+	var labels *apiv2.Labels
+	if t.Meta != nil && t.Meta.Labels != nil && len(t.Meta.Labels) > 0 {
+		labels = &apiv2.Labels{
+			Labels: tag.NewTagMap(t.Meta.Labels),
+		}
+	}
+
+	return &apiv2.Tenant{
+		Login:       t.Meta.Id,
+		Name:        t.Name,
+		Description: t.Description,
+		Email:       t.Meta.Annotations[TenantTagEmail],
+		AvatarUrl:   t.Meta.Annotations[TenantTagAvatarURL],
+		CreatedBy:   t.Meta.Annotations[TenantTagCreator],
+		Meta: &apiv2.Meta{
+			CreatedAt: t.Meta.CreatedTime,
+			UpdatedAt: t.Meta.UpdatedTime,
+			Labels:    labels,
+		},
+	}, nil
+}
+
+func (r *tenantRepository) Member(tenantID string) TenantMember {
+	return r.tenantMember(&TenantScope{
+		tenantID: tenantID,
+	})
+}
+
+func (r *tenantRepository) UnscopedMember() TenantMember {
+	return r.tenantMember(nil)
+}
+
+func (r *tenantRepository) tenantMember(scope *TenantScope) TenantMember {
+	repository := &tenantMemberRepository{
+		s:     r.s,
+		scope: scope,
+	}
+
+	return &store[*tenantMemberRepository, *mdcv1.TenantMember, *mdcv1.TenantMember, *TenantMemberCreateRequest, *TenantMemberUpdateRequest, *TenantMemberQuery]{
+		repository: repository,
+		typed:      repository,
+	}
 }
 
-// Update implements Tenant.
-func (t *tenantRepository) update(ctx context.Context, e *mdcv1.Tenant, msg *apiv2.TenantServiceUpdateRequest) (*mdcv1.Tenant, error) {
-	panic("unimplemented")
+func (t *tenantRepository) ListTenantMembers(ctx context.Context, tenant string, includeInherited bool) ([]*mdcv1.TenantWithMembershipAnnotations, error) {
+	resp, err := t.s.mdc.Tenant().ListTenantMembers(ctx, &mdcv1.ListTenantMembersRequest{
+		TenantId:         tenant,
+		IncludeInherited: pointer.Pointer(includeInherited),
+	})
+	if err != nil {
+		return nil, errorutil.Convert(err)
+	}
+
+	return resp.Tenants, nil
+}
+
+func (r *tenantRepository) FindParticipatingTenants(ctx context.Context, tenant string, includeInherited bool) ([]*mdcv1.TenantWithMembershipAnnotations, error) {
+	resp, err := r.s.mdc.Tenant().FindParticipatingTenants(ctx, &mdcv1.FindParticipatingTenantsRequest{
+		TenantId:         tenant,
+		IncludeInherited: pointer.Pointer(includeInherited),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Tenants, nil
 }
 
-// ConvertToInternal implements Tenant.
-func (t *tenantRepository) convertToInternal(msg *apiv2.Tenant) (*mdcv1.Tenant, error) {
-	panic("unimplemented")
+func TenantRoleFromMap(annotations map[string]string) apiv2.TenantRole {
```

Comment:
unimport

Reply:
For this function it does not work because it's used in the services, but it was possible to do so for `ProjectRoleFromMap`.